app/models/user.rb

```diff
@@ -48,4 +52,17 @@ def resume_attachment
     self.resumes.where.not( resume_attachment_file_name: nil ).last.resume_attachment.url
   end
 
+  def has_faved_the_company?(company)
+    self.fav_company_ids.include?(company.id)
+  end
+
+  def fav_company(company, user)
+    UserFavCompany.create( company: company, user: user )
+  end
```

Comment:
def fav_company(company)
     UserFavCompany.create( company: company, user: self )
  end

is better